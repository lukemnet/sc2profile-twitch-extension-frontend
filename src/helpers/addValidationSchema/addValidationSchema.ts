import { YupInstance } from 'yup';
import StarCraft2API from 'starcraft2-api';
import isProfileListUnique from 'helpers/isProfileListUnique/isProfileListUnique';
import phrases from 'config/phrases';

const addValidationSchema = (yup: YupInstance) => ({
  profiles: yup.array()
    .of(
      yup.string()
        .ensure()
        .min(34, phrases.FORM_VALIDATION_URL_TOO_SHORT)
        .matches(
          StarCraft2API.profileUrlRegex(),
          {
            message: phrases.FORM_VALIDATION_URL_INVALID,
            excludeEmptyString: false,
          },
        )
    )
    .unique(phrases.FORM_VALIDATION_DUPLICATE_URLS, isProfileListUnique),
});

export default addValidationSchema;
