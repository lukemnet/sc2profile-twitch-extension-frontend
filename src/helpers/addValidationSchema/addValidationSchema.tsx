import { YupInstance } from 'yup';
import StarCraft2API from 'starcraft2-api';

const addValidationSchema = (yup: YupInstance) => ({
  profiles: yup.array()
    .of(
      yup.string()
        .ensure()
        .min(34, 'too short')
        .matches(StarCraft2API.profileUrlRegex(),
          {
            message: 'invalid SC2 profile',
            excludeEmptyString: false,
          },
        )
    )
    .unique('duplicate phone', (a: string) => a),
});

export default addValidationSchema;
