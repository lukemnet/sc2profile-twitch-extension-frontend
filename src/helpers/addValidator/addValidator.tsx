import getValidatorInstance from 'src/helpers/getValidatorInstance/getValidatorInstance';
import addUniqueProfileListValidationMethod from 'src/helpers/addUniqueProfileListValidationMethod/addUniqueProfileListValidationMethod';

const addValidator = () => {
  const validatorInstance = getValidatorInstance();
  return addUniqueProfileListValidationMethod(validatorInstance);
};

export default addValidator;
