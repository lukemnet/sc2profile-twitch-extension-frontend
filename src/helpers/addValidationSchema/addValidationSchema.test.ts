import addValidator from '../addValidator/addValidator';
import addValidationSchema from './addValidationSchema';

it('returns valid validation schema', () => {
  const validator = addValidator();
  const validationSchema = addValidationSchema(validator);
  expect(validationSchema).toMatchSnapshot();
});
