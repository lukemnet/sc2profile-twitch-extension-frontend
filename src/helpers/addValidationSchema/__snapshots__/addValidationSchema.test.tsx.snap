// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns valid validation schema 1`] = `
Object {
  "profiles": ArraySchema {
    "_blacklist": RefSet {
      "list": Set {},
      "refs": Map {},
    },
    "_conditions": Array [],
    "_deps": Array [],
    "_exclusive": Object {
      "unique": false,
    },
    "_mutate": undefined,
    "_options": Object {
      "abortEarly": true,
      "recursive": true,
    },
    "_subType": StringSchema {
      "_blacklist": RefSet {
        "list": Set {},
        "refs": Map {},
      },
      "_conditions": Array [],
      "_default": "",
      "_deps": Array [],
      "_exclusive": Object {
        "min": true,
        "undefined": false,
      },
      "_mutate": undefined,
      "_options": Object {
        "abortEarly": true,
        "recursive": true,
      },
      "_type": "string",
      "_typeError": [Function],
      "_whitelist": RefSet {
        "list": Set {},
        "refs": Map {},
      },
      "tests": Array [
        [Function],
        [Function],
      ],
      "transforms": Array [
        [Function],
        [Function],
      ],
    },
    "_type": "array",
    "_typeError": [Function],
    "_whitelist": RefSet {
      "list": Set {},
      "refs": Map {},
    },
    "tests": Array [
      [Function],
    ],
    "transforms": Array [
      [Function],
    ],
  },
}
`;
